File: Ruby/problem002.rb
#! /usr/bin/env ruby
=begin
Problem 2 - Project Euler
http://projecteuler.net/index.php?section=problems&id=2
=end

def fibevensum xmax
    a, b = 1, 2
    sum = 0
    while a < xmax
        sum += a if a.even?
        a, b = b, a + b
    end
    sum
end

puts fibevensum 4_000_000

File: Swift/problem002.swift
// Problem 2 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=2
func fibevensum(n: Int) -> Int {
    var a = 1
    var b = 2
    var sum = b
    var tmp = a

    while a < n {
        tmp = b
        b = a + b
        a = tmp

        if b % 2 == 0 {
            sum += b
        }
    }
    return sum
}

print(fibevensum(n: 4_000_000))
